Add support for `MintTokensTx` and `SetAccountDelegateTx`

jordipainan opened the pull request.

jordipainan committed:
vochain: add support for MintTokensTx

Added support for MintTokensTx.
Change Treasurer from hex address to
protobuf Treasurer model and adapt
the existing code accordingly.

jordipainan committed:
vochain: ethless add support for add/del account delegates

jordipainan committed:
proto: bump version

jordipainan committed:
vochain: add/del account delegates

Increase nonce on add/del success.
Adds unit testing.

p4u commented on vochain/account.go:
```diff
@@ -282,3 +284,124 @@ func SetAccountInfoTxCheck(vtx *models.Tx, txBytes, signature []byte, state *Sta
+		for i := 0; i < len(acc.DelegateAddrs); i++ {
+			delegateToCmp := common.BytesToAddress(acc.DelegateAddrs[i])
+			if delegateToCmp == delAcc {
+				f = true
```
just put the return here `return sigAddress, delAcc, nil` remove `f`and return by default `return common.Address{}, common.Address{}, fmt.Errorf("cannot remove a non existent delegate")`

p4u commented on vochain/state.go:
```diff
@@ -503,38 +503,63 @@ func (v *State) IsOracle(addr common.Address) (bool, error) {
+		Address: address.Bytes(),
+		Nonce:   0,
+	}
+	tBytes, err := proto.Marshal(t)
```
```
tBytes, err := proto.Marshal(&models.Treasurer{
		Address: address.Bytes(),
		Nonce:   0,
	}
```

randomshinichi committed:
VPROT-125 TransactionCosts{} represents the tx cost settings in the genesis file (#398)

add transaction costs to the genesis

jordipainan committed:
address minor comments:
- simplify setTreasurer
- delete unrequired boolean from SetAccountDelegateTxCheck()
